app/main/views/service_updates.py

```diff
@@ -54,33 +56,24 @@ def service_status_update_audits(day=None, page=1):
 @login_required
 @role_required('admin', 'admin-ccs-category')
 def service_update_audits():
-    form = ServiceUpdateAuditEventsForm(request.args, csrf_enabled=False)
-
-    if not form.validate():
-        return render_template(
-            "service_update_audits.html",
-            today=datetime.utcnow().strftime(DATETIME_FORMAT),
-            acknowledged=form.default_acknowledged(),
-            audit_events=[],
-            form=form
-        ), 400
-
-    audit_events = data_api_client.find_audit_events(
+    audit_events = list(data_api_client.find_audit_events_iter(
```

Comment:
So, yes, this does indeed start to feel like using an api that wasn't designed for this, but then again specializing down into a set of endpoints for "service updates" starts to go in the direction of "backend has special purpose endpoint for each frontend endpoint" which is very... non-api-y. I *think* simply adding a "first for each object" option on the listing endpoint would achieve all we need for this view and might have uses elsewhere.

When it comes to the whole "people making edits faster than we can keep up with them", I assume we're going to have to put up with the idea of some edits being live for a couple of days anyway what with CCS having weekends. And a couple of days is enough for us to fix anything (right?).

I was thinking about being able to pre-test the setup and reckoned the cheapest/easiest way to do it for now for lack of a similar testing setup would be to roll out some conservative api changes to production and test e.g. hitting it ten times in a row. We *do have* some service update events in the log at the moment to give us an example dataset - I don't think any database writes would be needed.